src/test/kotlin/com/sksamuel/avro4k/schema/RecursiveSchemaTest.kt

```diff
@@ -0,0 +1,39 @@
+package com.sksamuel.avro4k.schema
+
+import com.sksamuel.avro4k.Avro
+import com.sksamuel.avro4k.io.AvroInputStream
+import com.sksamuel.avro4k.io.AvroOutputStream
+import io.kotest.matchers.shouldBe
+import io.kotest.core.spec.style.FunSpec
+import kotlinx.serialization.Serializable
+import java.io.File
+
+@Serializable
+data class Recursive(val payload: Int, val next: Recursive?)
+
+@Serializable
+data class Level4(val level1: Level1)
+
+@Serializable
+data class Level3(val level4: Level4?)
+
+@Serializable
+data class Level2(val level3: Level3)
+
+@Serializable
+data class Level1(val level2: Level2?)
+
+class RecursiveSchemaTest : FunSpec({
```

Comment:
Can you also add test cases in which the type is referred through a collection type?

e.g.:
```kotlin
@Serializable
data class RecursiveListItem(val payload: Int, val allItems: List<RecursiveListItem>)
```
This would also be needed for Array, Map and Pair. I would guess, just by reading the code, that these tests would fail right now.

Reply:
Nice catch